MyString.java

```diff
@@ -36,8 +47,14 @@ public static int countChar(String str, char ch) {
      * @return true is str1 is a subset of str2, false otherwise
      */
     public static boolean subsetOf(String str1, String str2) {
-         //// Replace the following statement with your code
-        return false;
+        if (str1.length() == 0) return true; 
+        if (str1.length() > str2.length()) return false; 
+        for (int i = 0; i < str1.length(); i++) {
+            if (countChar(str2, str1.charAt(i))==0 || countChar(str2, str1.charAt(i)) < countChar(str1, str1.charAt(i))) {

```

Comment:
redundant case `countChar(str2, str1.charAt(i))==0`, always will be caught in second condition
since in that case
`countChar(str2, str1.charAt(i)) = 0 `
`countChar(str1, str1.charAt(i)) = n, such that n >= 1;`
hence 
`countChar(str1, str1.charAt(i))) > countChar(str2, str1.charAt(i))`